### ISSUE TITLE
[Feature Request] Add option to hide 0-valued series from tooltip when Tooltip Mode = All series

### ISSUE THREAD

**Issue by toddwh50** [grafana/grafana#4623]
When graph is comprised of "sparser" data where some times do not have values for all series (see image), it would be useful to be able to check a "Hide 0-Valued Series in Tooltip" option to cause these to be excluded from the tooltip. Then, only series having values would remain.

![image](https://cloud.githubusercontent.com/assets/17885456/14398537/89c207d6-fdaa-11e5-85f3-58e843cf3fce.png)

**Comment by scraly** [comment]
Hi,
I still have the issue with grafana 3.1.0 version for "All series" tooltip mode.
![grafana_tooltip_0_issue](https://cloud.githubusercontent.com/assets/2543381/17176145/9bf3e07a-540b-11e6-8a79-51f82b4017b6.png)

**Comment by mrares** [comment]
+1

**Comment by ldktta** [comment]
+1

**Comment by scraly** [comment]
+1

**Comment by scraly** [comment]
![grafana_0_value_402](https://cloud.githubusercontent.com/assets/2543381/21257083/3abe33d4-c376-11e6-8da2-d8794c5cd7b8.png)
Problem exists again and again :-).

**Comment by scraly** [comment]
Hi, is it possible to have a look at this? Thanks

**Comment by mrares** [comment]
I "solved" it by sorting the tooltip, any >0 values appear on top... still the UI is bad.

**Comment by ivanpricewaycom** [comment]
sorting desc by tooltip is kinda ok, but very large tooltips can run above the window. it seems like this should be fairly easy to 'fix' around line ~130 of this file:

https://github.com/grafana/grafana/blob/8b25ba6a7c8c26191704ddfe77dec524072cf13c/public/app/plugins/panel/graph/graph_tooltip.js

but i can't work out how to rebuild just the JS, any tips / pointers ?

**Comment by james-johnston-thumbtack** [comment]
+1

**Comment by scraly** [comment]
Any news? :-)

**Comment by jaune-rouge** [comment]
+1

**Comment by amiralibazazi** [comment]
+1

**Comment by jdefaver** [comment]
+1

**Comment by starcraft66** [comment]
+1

**Comment by machytkafitanalytics** [comment]
+1

**Comment by ebolstad** [comment]
+1

**Comment by oleg-glushak** [comment]
+1

**Comment by lambdaq** [comment]
same here. I try not to type +1 in plain, but still, +1.

**Comment by sunilkukde** [comment]
+1

**Comment by pukseli** [comment]
+1

**Comment by akuma** [comment]
+1

**Comment by daveselan** [comment]
Any updates on this?

**Comment by ChrisScottThomas** [comment]
+1

**Comment by SmetDenis** [comment]
+2

**Comment by 1337blue** [comment]
+1 BTW chronograf does this oob

**Comment by herlon214** [comment]
+1

**Comment by gkrasulya** [comment]
+1

**Comment by tfalbo** [comment]
+1

**Comment by gliu93** [comment]
+1

**Comment by RPCMoritz** [comment]
This is really annoying with stacked graphs, where you hover over the line of a non-zero area, but with the single-tooltip setting activated, you get another serie's 0-value at that time, and you may not be able to recover the value that actually corresponds to the increased value. If activate the all-tooltip setting, you still have to color/value match, and hope that you're able to zoom in far enough, that you can exclude a sufficient number of data points, to not get massive tooltip pollution.

**Comment by JavierHernandezMontes** [comment]
This feature request was opened 5 years ago.. And still no answer...

**Comment by pukseli** [comment]
> This feature request was opened 5 years ago.. And still no answer...

Yeah. I used to be a summer trainee for the 1st summer when I created this issue. Now I'm almost graduated and have a regular full time job. Time flies.

**Comment by davidmehren** [comment]
This is still a vald feaure request.

**Comment by gelicia** [comment]
The team chatted about this one - we will plan to implement this by adding a "Hide Zeros" true/false option to the tool tips options.
